controller/share_manager_controller.go

```diff
@@ -405,27 +405,51 @@ func (c *ShareManagerController) isShareManagerRequiredForVolume(sm *longhorn.Sh
 	return false
 }
 
-func (c *ShareManagerController) createShareManagerAttachmentTicket(sm *longhorn.ShareManager, va *longhorn.VolumeAttachment) {
+func (c *ShareManagerController) createShareManagerAttachmentTicket(sm *longhorn.ShareManager, va *longhorn.VolumeAttachment) error {
 	log := getLoggerForShareManager(c.logger, sm)
+	podName := types.GetShareManagerPodNameFromShareManagerName(sm.Name)
+	pod, err := c.ds.GetPodRO(c.namespace, podName)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return errors.Wrapf(err, "failed to retrieve pod %v for share manager from datastore", podName)
+	}
+
+	// This is not a fatal error, just wait for the pod.
+	if pod == nil {
+		log.Infof("No share-manager pod yet to attach the volume %v", va.Name)
+		return nil
+	}
+	if pod.Spec.NodeName == "" {
+		log.Infof("Share-manager pod %v for volume %v reports no owner node", pod.Name, va.Name)
+		return nil
+	}
+	if pod.Status.Phase != corev1.PodPending && pod.Status.Phase != corev1.PodRunning {
+		log.Infof("Share-manager pod %v for volume %v reports unusable phase %v", pod.Name, va.Name, pod.Status.Phase)
+		return nil
+	}
```

Comment:
nit: Suggest adding the code about pod (pre-)readness in a common place for reuse.

Reply:
Will do.  I'll save this for the next set of changes.